# SCRIPpro
The increasing availability of single-cell RNA-seq data allows characterizing gene regulation networks (GRNs) at an unprecedented resolution. Previously we developed SCRIP, a computational method that integrates single-cell ATAC-seq data with a large-scale transcription regulator (TR) ChIP-seq data and motif reference for reconstructing single-cell TR activities and GRNs. Here, we present SCRIPpro, an extended framework of SCRIP that suits both single-cell and spatial RNA-seq data. SCRIPpro first applied a pre-clustering strategy using Supercell to increase the coverage of scRNA-seq data. Then SCRIPpro identifies the marker genes of each Supercell cluster and utilize the provided scATAC-seq data or Dnase/H3k27ac sequencing data from cistrome as chromatin landscape to do in silico deletion , and using TR targets deletion to evaluate the importance of each TR in Supercell based on a large-scale TR reference. Finally, SCRIPpro combines the gene expression of each TR with its importance score using GRNBOOST2, which generates the TR-centered GRNs at Supercell resolution. We applied SCRIPpro on human PBMC scRNA-seq data and mouse developing embryo spatial transcriptomic dataset , and demonstrated that SCRIPpro is able to identify cell-type-specific TR regulations and show superior performance compared to conventional motif-based methods such as SCENIC. Taken together, SCRIPpro is a convenient and fast method that could reconstruct TR activities and GRNs for both scRNA-seq and spatial transcriptomic data.

![image](https://github.com/xuyunfan9991/SCRIPpro/assets/143389390/cd7774ea-23ba-45d5-8923-1bca61cf148f)


